kontrol/keypair.go

```diff
@@ -47,6 +47,10 @@ type KeyPairStorage interface {
 
 	// GetKeyFromPublic retrieves the KeyPairs from the given public Key
 	GetKeyFromPublic(publicKey string) (*KeyPair, error)
+
+	// Is valid checks if the given publicKey is valid or not. It's up to the
+	// implementer how to implement it. A valid public key returns a nil error.
+	IsValid(publicKey string) error
```

Comment:
same functionality can be handled with `GetKeyFromPublic` function if it returns and exported `ErrNotFound` error.